conformance/tests/admin-network-policy-core-ingress-tcp-rules.go

```diff
@@ -56,10 +56,19 @@ var AdminNetworkPolicyIngressTCP = suite.ConformanceTest{
 				Name:      "harry-potter-0",
 			}, serverPod)
 			require.NoErrorf(t, err, "unable to fetch the server pod")
+
+			// ensure ingress is ALLOWED from dumbledore and DENIED from professor quirrell. Both pods are in the hogwarts-staff namespace
+			success := kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-hogwarts-staff", "professor-dumbledore-0", "tcp",
+				serverPod.Status.PodIP, int32(80), s.TimeoutConfig.RequestTimeout, true)
+			assert.Equal(t, true, success)
+			success = kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-hogwarts-staff", "professor-quirrell-0", "tcp",
```

Comment:
Not related to this PR, but I think we should consider refactoring the `PokeServer` function so that it is more readable. In particular, `shouldConnect` works better as a return type rather than a parameter:
```
success, connected :=  kubernetes.PokeServer(t, s.ClientSet, &s.KubeConfig, "network-policy-conformance-hogwarts-staff", "professor-dumbledore-0", "tcp", serverPod.Status.PodIP, int32(80), s.TimeoutConfig.RequestTimeout)
assert.True(t, success)
assert.True(t, connected) // or assert.False if expect the connection to be dropped
```
@tssurya

Reply:
It's funny you say that, because this function tripped me up the first time I ran my tests (because of said readbility), I incorrectly assumed the nature of it (assumed it it was intended to expect a false return on DENY and assert as such) and had to change my boolean values when writing the DENY examples once I looked at the function definition/params and understood it better.

So yeah 👍🏼 / +1, agree

Reply:
There is an issue I had opened that outlines the "we should refactor pokeServer" :) I admit when I initially added it, it was just a starter and deff needs enhancement in terms of readability. +1 from me as well

Reply:
https://github.com/kubernetes-sigs/network-policy-api/issues/108